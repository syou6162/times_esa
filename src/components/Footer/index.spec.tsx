import.meta.env.VITE_VALID_MAIL_ADDRESSES = 'valid@example.com';

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Footer, FooterProps } from '.'

describe('Footerが正しく表示される', () => {
  it('サインインしていない状態', () => {
    const props: FooterProps = {
      isSignedIn: false,
      user: {
        email: '',
        displayName: '',
        photoURL: '',
      },
      firebaseAuth: null
    }
    const renderResult = render(
      <Footer {...props} />
    );

    expect(renderResult.asFragment()).toMatchSnapshot();
  });
  it('invalidなユーザーでログインしている状態', () => {
    const props: FooterProps = {
      isSignedIn: true,
      user: {
        email: 'invalid@example.com',
        displayName: 'invalid',
        photoURL: 'invalid',
      },
      firebaseAuth: null
    }
    const renderResult = render(
      <Footer {...props} />
    );

    expect(renderResult.asFragment()).toMatchSnapshot();
  });
  it('validなユーザーでログインしている状態', () => {
    const props: FooterProps = {
      isSignedIn: true,
      user: {
        email: 'valid@example.com',
        displayName: 'valid',
        photoURL: 'valid',
      },
      firebaseAuth: null
    }
    const renderResult = render(
      <Footer {...props} />
    );

    expect(renderResult.asFragment()).toMatchSnapshot();
  });
})
