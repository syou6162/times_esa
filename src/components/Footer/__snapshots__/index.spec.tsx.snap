// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footerが正しく表示される invalidなユーザーでログインしている状態 1`] = `
<DocumentFragment>
  <footer>
    <hr
      style="border-top: 2px dashed #bbb; clear: both;"
    />
    <div>
      <a
        href="https://github.com/syou6162/times_esa"
      >
        times_esa
      </a>
       made by 
      <a
        href="https://twitter.com/syou6162"
      >
        Yasuhisa Yoshida
      </a>
      .
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Footerが正しく表示される validなユーザーでログインしている状態 1`] = `
<DocumentFragment>
  <footer>
    <hr
      style="border-top: 2px dashed #bbb; clear: both;"
    />
    <div>
      ようこそ valid (
      <img
        alt="valid"
        height="20px"
        src="valid"
        title="valid@example.com"
      />
      valid@example.com) さん!
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Sign-out
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <hr
      style="border-top: 2px dashed #bbb; clear: both;"
    />
    <div>
      <a
        href="https://github.com/syou6162/times_esa"
      >
        times_esa
      </a>
       made by 
      <a
        href="https://twitter.com/syou6162"
      >
        Yasuhisa Yoshida
      </a>
      .
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Footerが正しく表示される サインインしていない状態 1`] = `
<DocumentFragment>
  <footer>
    <hr
      style="border-top: 2px dashed #bbb; clear: both;"
    />
    <div>
      <a
        href="https://github.com/syou6162/times_esa"
      >
        times_esa
      </a>
       made by 
      <a
        href="https://twitter.com/syou6162"
      >
        Yasuhisa Yoshida
      </a>
      .
    </div>
  </footer>
</DocumentFragment>
`;
